cfgov/teachers_digital_platform/README_SURVEY.md

```diff
@@ -0,0 +1,120 @@
+
+In accordance with best accessibility practices, modal dialogs “trap” keyboard focus within the dialog contents, return focus to the element that opened the dialog when closed, and are given appropriate markup ARIA attributes to denote a dialog.
+
+On pages with modals, the initial modal markup (e.g. [privacy](jinja2/teachers_digital_platform/survey/modals/privacy.html)) is placed within the page and is hidden with CSS until an opener element is clicked.
+
+## Development
+
+The survey will work after a standard installation of the CFPB site, though surveys must be started by browsing to each grade's start page. E.g. [Grades 3-5 on localhost:8000](http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/survey/3-5/). The "reset" links within navigate to a Wagtail page which you may not have in your database.
+
+### Templates
+
+This gives an overview of template locations:
+
+| Desc               | URL                                           | Template in `jinja2/tea.../survey` | Script                                  |
+|--------------------|-----------------------------------------------|------------------------------------|-----------------------------------------|
+| Entry page for 3-5 | ../survey/3-5/                                | grade-level-3-5.html               | survey/grade-level-page.js              |
+| Survey page        | ../survey/3-5/p1/<br>../survey/3-5/p2/<br>... | form-page.html                     | survey/survey-page.js                   |
+| Results page       | ../survey/3-5/results/                        | results-3-5.html                   | survey/result-page.js                   |
+| Results page: car  | ../survey/3-5/results/                        | results-graph.html                 |                                         |
+| Privacy modal      |                                               | modals/privacy.html                | modals.js<br>survey/grade-level-page.js |
+
+Django's forms are customized to match CFPB markup guidelines more closely using a few Django HTML templates in [templates/](templates/teachers_digital_platform).
+
+### JavaScript
+
+JavaScript is split into 3 top-level modules (`grade-level-page`, `survey-page`, and `result-page`) to handle the different page types and a "router" component ([tdp-surveys](../unprocessed/apps/teachers-digital-platform/js/tdp-surveys.js)) executes the function in the appropriate module based on an HTML element with the attribute `data-tdp-page`.
+
+### Editing content
+
+The most reliable way to edit content is to first edit the [content Google Sheet](https://docs.google.com/spreadsheets/d/13wuLSJDBdelod3WZuOos5n-vcoUdQ66HYsdcM-az-y4/edit#gid=1402247033), then download the altered sheets as CSV files and replace them in the [survey-data](survey-data) folder.
```

Comment:
@mrclay I recommend losing the link to the Google Sheet. Maybe we can copy it to Sharepoint and have Hannah mention it in our internal documentation.